core/new-gui/src/app/workspace/component/navigation/navigation.component.html

```diff
@@ -1,13 +1,26 @@
 <div class='texera-navigation-body'>
   <div class='texera-navigation-padded'>
     <div class='texera-navigation-title'>
-      <a href="#">
-        Texera <span class="version-number">0.5.1</span><span class="git-commit-hash"> {{gitCommitHash}}</span>
+      <a href="#" class = "title-link">
+          <span class="title">Texera</span>
+          <span class="version-number">0.5.1</span>
       </a>
+      <div class="git-commit-hash-label">
+        revision:
+      </div>
+      <div class="git-commit-hash">
+        {{gitCommitHash}}
+      </div>
     </div>
+
+    <div class="texera-navigation-workflow-name" *ngIf="userSystemEnabled">
+      <span class="name-input" #nameInput role="textbox" contenteditable="true" spellcheck="false" placeholder="Untitled Workflow" (input)="onWorkflowNameChange($event)" ></span>
```

Comment:
I think it's good to just use the old two-way data binding. Is there something wrong with it?
`<input (change)="onWorkflowNameChange()" [(ngModel)]="currentWorkflowName"`

Reply:
agreed, @MysteriousChallenger ?

Reply:
https://github.com/Texera/texera/pull/923#discussion_r540591787